packages/blockchain-wallet-v4/src/types/Wallet.js

```diff
@@ -549,21 +551,30 @@ export const getHDPrivateKeyWIF = curry(
             xpriv
           )
         )
-        .map(xp => derivePrivateKey(network, xp, chain, index).keyPair.toWIF())
+        .map(xp => {
+          const node = derivePrivateKey(
+            network,
+            xp,
+            chain,
+            index
+          ).keyPair.toWIF()
+          const keypair = Bitcoin.ECPair.fromPrivateKey(node.privateKey)
```

Comment:
Maybe I'm not getting it, but this doesn't seem right.

Reply:
Yes, it looks like I have an unnecessary keypair instantiation-- will remove and revert back to original code here.

Reply:
The reason I am explicitly having to define an ECPair fromPrivateKey,  is because bip32 class no longer exposes a keypair attribute.

This portion of the code is broken (I had not tested this with second password).

It should look like the if statement after it.

```
.map(xp => {
  const node = derivePrivateKey(network, xp, chain, index)
  return Bitcoin.ECPair.fromPrivateKey(node.privateKey).toWIF()
})
```